devito/passes/clusters/cse.py

```diff
@@ -156,53 +160,59 @@ def _compact_temporaries(exprs, exclude):
 
 
 @singledispatch
-def count(expr):
+def count(expr, conds):
     """
     Construct a mapper `expr -> #occurrences` for each sub-expression in `expr`.
     """
     mapper = Counter()
     for a in expr.args:
-        mapper.update(count(a))
+        mapper.update(count(a, None))
```

Comment:
Is it possible to one-line these loops? `Counter(count(a, None) for a in expr.args)`